devtools/common/src/test/java/io/quarkus/cli/commands/AddExtensionsTest.java

```diff
@@ -64,6 +64,38 @@ void testPartialMatches() throws IOException {
         hasDependency(model, "quarkus-jdbc-postgresql");
     }
 
+    @Test
+    void testRegexpMatches() throws IOException {
+        File pom = new File("target/extensions-test", "pom.xml");
+
+        CreateProjectTest.delete(pom.getParentFile());
+        new CreateProject(new FileProjectWriter(pom.getParentFile()))
+                .groupId("org.acme")
+                .artifactId("add-extension-test")
+                .version("0.0.1-SNAPSHOT")
+                .doCreateProject(new HashMap<>());
+
+        File pomFile = new File(pom.getAbsolutePath());
+        AddExtensionResult result = new AddExtensions(new FileProjectWriter(pomFile.getParentFile()))
+                .addExtensions(new HashSet<>(asList("SmallRye*")));
+
+        result.isUpdated();
+
+        Model model = MojoUtils.readPom(pom);
+        hasDependency(model, "quarkus-smallrye-reactive-messaging");
+        hasDependency(model, "quarkus-smallrye-reactive-streams-operators");
+        hasDependency(model, "quarkus-smallrye-opentracing");
+        hasDependency(model, "quarkus-smallrye-metrics");
+        hasDependency(model, "quarkus-smallrye-reactive-messaging-kafka");
+        hasDependency(model, "quarkus-smallrye-health");
+        hasDependency(model, "quarkus-smallrye-openapi");
+        hasDependency(model, "quarkus-smallrye-jwt");
+        hasDependency(model, "quarkus-smallrye-context-propagation");
+        hasDependency(model, "quarkus-smallrye-reactive-type-converters");
+        hasDependency(model, "quarkus-smallrye-reactive-messaging-amqp");
+        hasDependency(model, "quarkus-smallrye-fault-tolerance");
+    }
```

Comment:
We need to test with more complicated regexps as all the escapes are uncovered.
Also, we need to check with non matching regexp

Reply:
What does happen in the case of `**`?